File: bench/interp/bench_sortcb.py
import time, functools
n = 120000
base = [0]*n
seed = 12345
for i in range(n):
    seed = (seed * 48271) % 2147483647
    base[i] = seed % 1000000
t0 = time.perf_counter()
total = 0
for _ in range(6):
    arr = list(base)
    arr.sort(key=functools.cmp_to_key(lambda a, b: -1 if a < b else (1 if a > b else 0)))
    total += arr[0] + arr[n-1]
print("result: %d" % total)
print("elapsed: %d ms" % int((time.perf_counter()-t0)*1000))

File: bench/interp/bench_sortcb.js
const n = 120000;
const base = new Array(n);
let seed = 12345;
for (let i = 0; i < n; i++) {
    seed = (seed * 48271) % 2147483647;
    base[i] = seed % 1000000;
}
const t0 = Date.now();
let total = 0;
for (let p = 0; p < 6; p++) {
    const arr = base.slice();
    arr.sort((a, b) => a - b);
    total += arr[0] + arr[n - 1];
}
console.log("result: " + total);
console.log("elapsed: " + (Date.now() - t0) + " ms");
